src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -5781,7 +5781,14 @@ public static PopupMenu createDownloadMenu(@NotNull TargetedMSRun run)
             // request if the download opens on the same page.
             String timeout = "setTimeout(function(){location.href=that.href;},400);return false;";
 
-            onClickScript = "if(_gaq) {that=this; _gaq.push(['_trackEvent', 'SkyDocDownload', '" + run.getContainer().getPath() + "', '" + run.getFileName() + "']); " + timeout + "}";
+            // Wrap in try/catch as a hacky way to not care which GA variants are active
+
+            // Universal Analytics - remove after conversion to GA4 is complete
+            onClickScript = "try {that=this; _gaq.push(['_trackEvent', 'SkyDocDownload', " + PageFlowUtil.qh(run.getContainer().getPath()) + ", " + PageFlowUtil.qh(run.getFileName()) + "]); } catch (err) {}";
+            // GA4 variant
+            onClickScript += "try {gtag('event', 'SkyDocDownload', {path: " + PageFlowUtil.qh(run.getContainer().getPath()) + ", fileName: " + PageFlowUtil.qh(run.getFileName()) + "}); } catch(err) {}";
```

Comment:
One more question, do fileName and path have to be HTML-encoded here?

Reply:
They do (though they weren't before this change). That's why I chose qh() instead of just q()